domain/src/main/kotlin/no/nav/su/se/bakover/domain/vilkår/PensjonsVilkår.kt

```diff
@@ -0,0 +1,215 @@
+
+        override val erAvslag: Boolean = vurderingsperioder.any { it.resultat == Resultat.Avslag }
+
+        override val resultat: Resultat =
+            if (erInnvilget) Resultat.Innvilget else if (erAvslag) Resultat.Avslag else Resultat.Uavklart
+
+        override fun hentTidligesteDatoForAvslag(): LocalDate? {
+            return vurderingsperioder
+                .filter { it.resultat == Resultat.Avslag }
+                .map { it.periode.fraOgMed }
+                .minByOrNull { it }
+        }
+
+        override fun erLik(other: Vilkår): Boolean {
+            return other is Vurdert && vurderingsperioder.erLik(other.vurderingsperioder)
+        }
+
+        fun minsteAntallSammenhengendePerioder(): List<Periode> {
+            return vurderingsperioder.map { it.periode }.minsteAntallSammenhengendePerioder()
+        }
+
+        companion object {
+            fun tryCreate(
+                vurderingsperioder: Nel<VurderingsperiodePensjon>,
+            ): Either<KunneIkkeLagePensjonsVilkår, Vurdert> {
+                if (vurderingsperioder.harOverlappende()) {
+                    return KunneIkkeLagePensjonsVilkår.OverlappendeVurderingsperioder.left()
+                }
+                return Vurdert(vurderingsperioder.kronologisk()).right()
+            }
+
+            fun createFromVilkårsvurderinger(
+                vurderingsperioder: Nel<VurderingsperiodePensjon>,
+            ): Vurdert =
+                tryCreateFromVurderingsperioder(vurderingsperioder).getOrHandle { throw IllegalArgumentException(it.toString()) }
+
+            fun tryCreateFromVurderingsperioder(
+                vurderingsperioder: Nel<VurderingsperiodePensjon>,
+            ): Either<KunneIkkeLagePensjonsVilkår, Vurdert> {
+                if (vurderingsperioder.harOverlappende()) {
+                    return KunneIkkeLagePensjonsVilkår.OverlappendeVurderingsperioder.left()
+                }
+                return Vurdert(vurderingsperioder.kronologisk()).right()
+            }
+        }
+
+        sealed class UgyldigPensjonsVilkår {
+            object OverlappendeVurderingsperioder : UgyldigPensjonsVilkår()
+        }
+
+        override fun oppdaterStønadsperiode(stønadsperiode: Stønadsperiode): PensjonsVilkår {
+            check(vurderingsperioder.count() == 1) { "Kan ikke oppdatere stønadsperiode for vilkår med med enn èn vurdering" }
+            return copy(
+                vurderingsperioder = vurderingsperioder.map {
+                    it.oppdaterStønadsperiode(stønadsperiode)
+                },
+            )
+        }
+
+        override fun slåSammenLikePerioder(): PensjonsVilkår {
+            return copy(vurderingsperioder = vurderingsperioder.slåSammenLikePerioder())
+        }
+    }
+}
+
+data class VurderingsperiodePensjon private constructor(
+    override val id: UUID = UUID.randomUUID(),
+    override val opprettet: Tidspunkt,
+    override val resultat: Resultat,
+    override val grunnlag: Pensjonsgrunnlag?,
```

Comment:
Siden vi har lagt ved grunnlag på denne, kan den vel være non-nullable her?

Reply:
Avventer endelig modell for grunnlaget.